library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -219,36 +240,23 @@ internal class TransactionPayloadFragment :
             if (type == TYPE_RESPONSE && responseBitmap != null) {
                 result.add(TransactionPayloadItem.ImageItem(responseBitmap))
             } else if (!isBodyPlainText) {
-                fragment.context?.getString(R.string.chucker_body_omitted)?.let {
+                context?.getString(R.string.chucker_body_omitted)?.let {
                     result.add(TransactionPayloadItem.BodyLineItem(SpannableStringBuilder.valueOf(it)))
                 }
             } else {
                 bodyString.lines().forEach {
                     result.add(TransactionPayloadItem.BodyLineItem(SpannableStringBuilder.valueOf(it)))
                 }
             }
-
-            return result
-        }
-
-        override fun onPostExecute(result: List<TransactionPayloadItem>) {
-            fragment.payloadBinding.apply {
-                loadingProgress.visibility = View.INVISIBLE
-                responseRecyclerView.visibility = View.VISIBLE
-                responseRecyclerView.adapter = TransactionBodyAdapter(result)
-            }
-            fragment.requireActivity().invalidateOptionsMenu()
+            return@withContext result
         }
     }
 
-    class FileSaverTask(private val fragment: TransactionPayloadFragment) :
-        AsyncTask<Triple<Int, Uri, HttpTransaction>, Unit, Boolean>() {
-
-        @Suppress("NestedBlockDepth")
-        override fun doInBackground(vararg params: Triple<Int, Uri, HttpTransaction>): Boolean {
-            val (type, uri, transaction) = params[0]
+    @Suppress("NestedBlockDepth", "ThrowsCount")
+    private suspend fun saveToFile(type: Int, uri: Uri, transaction: HttpTransaction): Boolean {
+        return withContext(Dispatchers.IO) {
             try {
-                val context = fragment.context ?: return false
+                val context = context ?: return@withContext false
```

Comment:
Let's remove this `val` and use `requireContext` on the next line.

Reply:
👍